qc_tests/streaks.py

```diff
@@ -93,102 +170,354 @@ def get_repeating_string_threshold(obs_var, config_dict, plots=False, diagnostic
 
-            compressed_flags[start : end] = "K"
+    # flag identified streaks
+    for streak in bad:
+        start = int(np.sum(grouped_diffs[:streaks[streak], 1]))
+        end = start + int(grouped_diffs[streaks[streak], 1]) + 1
 
-            if plots:
-                plot_streak(times, this_var, start, end)
+        compressed_flags[start : end] = "K"
 
-        # undo compression and write into original object (the one with calm periods)
-        flags[this_var.data.mask == False] = compressed_flags
-        obs_var.flags = utils.insert_flags(obs_var.flags, flags)
+        if plots:
+            plot_streak(masked_times, this_var.data, obs_var.units, start, end)
+
+    # undo compression and write into original object (the one with calm periods)
+    flags[this_var.data.mask == False] = compressed_flags
+    obs_var.flags = utils.insert_flags(obs_var.flags, flags)
 
-    logger.info(f"Repeated Strings {this_var.name}")
+    logger.info(f"Repeated streaks {this_var.name}")
     logger.info(f"   Cumulative number of flags set: {len(np.where(flags != '')[0])}")
 
     return # repeating_value
 
 
 #************************************************************************
-def excess_repeating_value():
+def excess_repeating_value(obs_var: utils.Meteorological_Variable, times: np.array,
+                    config_dict: dict, 
+                    wind_speed: Optional[utils.Meteorological_Variable] = None,
+                    plots: bool = False, diagnostics: bool = False) -> None:
+    """
+    Flag years where more than expected fraction of data occurs in streaks,
+      but none/not many are long enough in themselves to trigger the repeating_value check
+      Themselves need to be a reasonable length (10) else this overflags
+
+    Use config file to read threshold values.  Then find streaks which exceed threshold.
+
+    :param MetVar obs_var: meteorological variable object
+    :param array times: array of times (usually in minutes)
+    :param str config_dict: configuration file to store critical values
+    :param MetVar wind_speed: need speeds to mask calm periods in wind_directions
+    :param bool plots: turn on plots
+    :param bool diagnostics: turn on diagnostic output
+    
+    """
+    years = np.array([t.year for t in times])
+    if plots:
+        # Needed for plotting only
+        masked_times = np.ma.array(times, mask=obs_var.data.mask)
+    
+    this_var = prepare_obs_var(obs_var, wind_speed=wind_speed)
+
+    # use mask rather than compress immediately, to ensure indexing works
+    flags = np.ma.array(["" for i in range(this_var.data.shape[0])],
+                        mask=this_var.data.mask)
+
+    # retrieve the threshold and store
+    try:
+        threshold = config_dict[f"STREAK-{this_var.name}"]["Excess"]
+    except KeyError:
+        # no threshold set
+        get_excess_streak_threshold(this_var, years, config_dict,
+                                    plots=plots, diagnostics=diagnostics)
```

Comment:
Check naming convention with excess repeated values or is it repeated streaks of values?

Reply:
I'll amend this naming once the test name is finalised.